src/wallet/wallet2.h

```diff
@@ -2415,6 +2530,29 @@ namespace boost
         return;
       a & x.multisig_sigs;
     }
+
+    template <class Archive>
+    inline void serialize(Archive& a, tools::wallet2::background_synced_tx &x, const boost::serialization::version_type ver)
+    {
+      a & x.index_in_background_sync_data;
+      a & x.tx;
+      a & x.output_indices;
+      a & x.height;
+      a & x.block_timestamp;
+      a & x.double_spend_seen;
+    }
```

Comment:
For my own understanding, what is happening here? Does `Acrhive` need to have its `operator&` overloaded for this to make sense? Is `a` being mutated via the & operator overload? I surmise that the & operator is being overloaded to mutate Archive's state because `a` is taken by reference and there doesn't seem to be a return value from `a & thing`

Reply:
https://stackoverflow.com/questions/71874042/how-does-boost-serialization-make-use-of-operator-to-implement-archiving-seri